### PR TITLE
Segment metadata query on tombstone

### PR DESCRIPTION
**Pull request by findingrish**

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java [hunk 1]
```diff
@@ -120,7 +123,6 @@ public ColumnHolder getColumnHolder(String columnName)
           {
             throw new UnsupportedOperationException();
```
Comment (abhishekrb19) [inline processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java:124]: Should this also return an empty or null column holder?

File: processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java [hunk 2]
```diff
@@ -93,13 +96,13 @@ public BitmapFactory getBitmapFactoryForDimensions()
...
           {
-            throw new UnsupportedOperationException();
+            return null;
           }
 
           @Override
           public Map<String, DimensionHandler> getDimensionHandlers()
           {
-            throw new UnsupportedOperationException();
+            return new HashMap<>();
```
Comment (clintropolis) [inline processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java:105]: maybe `Collections.emptyMap()` so it doesn't create garbage

File: processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java [hunk 3]
```diff
@@ -111,7 +114,7 @@ public void close()
           @Override
           public List<String> getColumnNames()
           {
-            throw new UnsupportedOperationException();
+            return new ArrayList<>();
```
Comment (clintropolis) [inline processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java:117]: Collections.emptyList()

File: processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java [hunk 4]
```diff
@@ -130,6 +132,24 @@ public boolean isFromTombstone()
...
+      @Override
+      public DateTime getMaxTime()
+      {
+        return tombstone.getInterval().getEnd();
+      }
+
+      @Override
+      public RowSignature getRowSignature()
+      {
+        return RowSignature.builder().build();
```
Comment (clintropolis) [inline processing/src/main/java/org/apache/druid/segment/loading/TombstoneSegmentizerFactory.java:151]: should create a static final empty signature to re-use for all tombstones

**Pull request closed without merging**